### PR TITLE
Set head according to fcuState even when attributes are incorrect

### PR DESCRIPTION
**Pull request by flcl42**
Fixes Closes Resolves [hive test](https://hivecancun.ethdevops.io/suite.html?suiteid=1695894242-926ed35efc158e6dbca870d4798848c9.json&suitename=engine-cancun#test-197)

## Changes

- Sets head according to fcuState even when attributes are incorrect

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V3.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using System.Threading;
 using System.Threading.Tasks;
 using FluentAssertions;
+using k8s.Models;
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V3.cs:11]: what is this?

Comment (flcl42) [reply src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V3.cs:11]: Just a useless ns added automatically, will not be in the final PR for sure

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs [hunk 2]
```diff
@@ -66,18 +69,29 @@ public class ForkchoiceUpdatedHandler : IForkchoiceUpdatedHandler
...
-    public Task<ResultWrapper<ForkchoiceUpdatedV1Result>> Handle(ForkchoiceStateV1 forkchoiceState, PayloadAttributes? payloadAttributes)
+    public Task<ResultWrapper<ForkchoiceUpdatedV1Result>> Handle(ForkchoiceStateV1 forkchoiceState, PayloadAttributes? payloadAttributes, int version)
+    {
+        Block? newHeadBlock = GetBlock(forkchoiceState.HeadBlockHash);
+        ResultWrapper<ForkchoiceUpdatedV1Result>? payloadUpdateResult = ApplyForkchoiceUpdate(newHeadBlock, forkchoiceState, payloadAttributes);
+        return Task.FromResult(
+            ValidateAttributes(payloadAttributes, version) ??
+            payloadUpdateResult ??
+            StartBuildingPayload(newHeadBlock!, forkchoiceState, payloadAttributes)
+        );
```
Comment (flcl42) [inline src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs:84]: This is the main change:
- run ApplyForkchoiceUpdate, 
- only then validate params and fork, return error if found, 
- or error from ApplyForkchoiceUpdate 
- or anything else returned by StartBuildingPayload)

@jflo @MarekM25

**Comment by LukaszRozmej** [comment]
This looks wrong to me; the wrong version is called and we are doing everything and only returning an error at the end. Sounds wrong. Did you consider that the hive test is poorly constructed and should be amended? Did you discuss it with Hive test guys? Did you discuss it as spec?

**Comment by flcl42** [comment]
> This looks wrong to me; the wrong version is called and we are doing everything and only returning an error at the end. Sounds wrong. 

Wrong version of what? It returns result if presents, result can be an error or SYNCING/VALID, not just error. Let's call?

> Did you consider that the hive test is poorly constructed and should be amended? Did you discuss it with Hive test guys? Did you discuss it as spec?

[Yes, sure, it was not obvious for me too](https://discord.com/channels/@me/1099036146387914853/1156949341961998428)

**Review by LukaszRozmej — APPROVED** [review]
I hate the spec and would like to change it, but the PR looks good

**Review by marcindsobczak — APPROVED** [review]

**Pull request merged**